src/app/GitUI/CommandsDialogs/BrowseDialog/FormGitCommandLog.cs

```diff
@@ -116,13 +116,25 @@ private void CommandCacheItems_SelectedIndexChanged(object sender, EventArgs e)
         {
             string command = (string)CommandCacheItems.SelectedItem;
 
-            if (GitModule.GitCommandCache.TryGet(command, out byte[]? cmdOut, out byte[]? cmdErr))
+            if (GitModule.GitCommandCache.TryGet(command, out string? cmdOut, out string? cmdErr))
             {
                 Encoding encoding = GitModule.SystemEncoding;
                 commandCacheOutput.Text =
-                    command +
+                    PrintableChars(command) +
                     "\n-------------------------------------\n\n" +
-                    EncodingHelper.DecodeString(cmdOut, cmdErr, ref encoding).Replace("\0", "\\0");
+                    PrintableChars(cmdOut) +
+                    "\n-------------------------------------\n\n" +
+                    PrintableChars(cmdErr);
+
+                static string PrintableChars(string? str)
```

Comment:
```suggestion
                static string? PrintableChars(string? str)
```

Reply:
In #11848. Will rebase this PR when merged.